---
title: "Wednesday morning session"
output:
  html_document: default
  pdf_document: default
  word_document: default
editor_options:
  chunk_output_type: console
---


## Welcome to Tuesday! 

In your group discuss:

- yesterday's questions (below); can you agree on the answers?
- anything that you've found confusing or useful yesterday
- and post your answers and comments on our [whiteboard](https://jamboard.google.com/d/1PSY_jTxLy26TsIFbmxnOLe-tP8LOSsK-AzES-ihx4qo/edit?usp=sharing)


-----

**Tuesday's questions**

1. Which of the following is a peak-independent QC method for ChIP-seq
a. clustering of binned signal in bam files
b. cross correlation
c. fraction of reads in peaks (FRiP)
d. fraction of reads mapped to mitochondrial chromosome

2. Replicate consistency in ChIP-seq can be assessed by
a. PCA of binned signal in BAM files
b. clustering of signal in peak regions
c. IDR
d. all the above

3. Which of the following does NOT apply to broad peaks
a. sequencing depth does not scale linearly with genome size
b. peak calling results are less reproducible between different methods
c. fragment length can be assessed by cross correlation
d. proper sequencing depth is crucial to obtaining meaningful results

4. Which of the following in NOT a downstream analysis for a standard ChIP-seq experiment
a. functional annotation of detected peaks
b. differential occupancy analysis
c. identification of sequence motifs in peaks
d. nucleosome positioning analysis

5. In ChIP-seq, the ideal IP and input pair should:
a. ChIP sample shows strong cross correlation at fragment length whereas input shows none 
b. ChIP sample and input group together when clustering is made based on signal in peak regions
c. the lines representing both look similar on the fingerprint plot
d. the input should show similar pattern of enrichment as the IP

6. To perform differential occupancy analysis one needs
a. quantify ChIP-seq signal as counts
b. many detected peaks in each sample
c. three independent biological replicates of each condition 
d. a and c

7. Which of the methods gives information on chromatin openness / accessibility
a. ChIP-seq against H3K4me1
b. ATAC-seq
c. HiC
d. ChIP-exo

8. Fragment length estimation is an important QC metric for ATAC-seq because it
a. is later used for peak calling
b. confirms that read mapping was succesful
c. demonstrates that mono- and di- nucleosome peaks are present in the data
d. it is not a QC metric for ATAC-seq experiments



