---
title: "Questions 03"
output:
  html_document: default
  pdf_document: default
  word_document: default
editor_options:
  chunk_output_type: console
---

1. Could you identify two internal normalization methods for traditional ChIP-Seq datasets?
a) RPKM (correct)
b) RPGC (correct)
c) TPM (incorrect, used for RNA-Seq)
d) division by input (incorrect, this corrects for uneven input but does not ensure normalization across samples)

2. Could you identify sources of background in ChIP-Seq?
a) unspecific binding of DNA to the surface of beads
b) chromatin fragments sticking together or remaining connected via crosslinks after sonication
c) antibody binding other proteins with lower affinity
d) all of the above (correct)

3. What is the underlying assumption of comparisons relying on internal ChIP-Seq normalization?
a) that the background between samples is constant (correct)
b) that total reads counts of the samples are the same
c) that the peak height-to-background ratio is constant
d) that inputs of the samples are 

4. How can a drug treatment change a histone PTM landscape?
a) new peaks appear
b) existing peaks get smaller or bigger
c) the background (in between peaks) increases or decreases
d) all of the above (correct)

5. Which factors should theoretically have no background in ChIP-Seq (i.e. binary signal, either peak or nothing)
a) DNA specific binding factors (correct)
b) RNA Polymerase
c) Chromatin remodelers
d) Lamin

6. How is Drosophila spike-in used for normalization?
a) each dataset is scaled so that all datasets match the same absolute spike-in read count (correct)
b) all datasets are pooled and and then normalized
c) each dataset is divided by its input
d) none of the above

7. How is it possible to assign reads to their sample when performing multiplexed ChIP (BarChIP, Mint-ChIP, MINUTE-ChIP)?
a) a DNA barcode is ligated onto chromatin fragments
b) proteins are tagged before pooling
c) barcoded RNA is added to each sample
d) IPs are performed separately

8. Could you identify ChIP-Seq alternatives for profiling TF binding?
a) CUT&Run, CUT&TAG, DamID (correct)
b) only CUT&Tag and CUT&TAG
c) CUT&RUN, CUT&TAG, DamID and ATAC-seq 
d) none of the above


Extra 

9. How can you establish if two or more datasets share the same TF binding sites?
a) Peak calling
b) Intersecting peaks (e.g. Venn diagramm)
c) Heatmaps of the joint peak lists
d) All of the above (correct)